pkgs/native_assets_builder/test_data/treeshaking_native_assets/lib/src/treeshaking_native_assets.dart

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+import 'treeshaking_native_assets_bindings.dart' as bindings;
+
+class MyMath {
+  @ResourceIdentifier('add')
+  static int add(int a, int b) => bindings.add(a, b);
+
+  @ResourceIdentifier('multiply')
```

Comment:
Can we address https://github.com/dart-lang/sdk/issues/55407 and https://github.com/dart-lang/native/issues/1085? (I'm happy to do the latter, but you can also do it yourself.) I'd feel more comfortable if this test exercises the pattern that we intend users to use.